### PR TITLE
add QUDA_MAGMAHOME cmake variable

### PR DESCRIPTION
**Pull request by kostrzewa**
to allow manual configuration of MAGMA directories
if not set, fall back to pkg-config

### CODE REVIEW

**Comment by cpviolator** [comment]
While you're doing this, can you change line 240 so that it reads as:
```cmake
message("${MAGMA_INCLUDEDIR}")
```
Without the quotes, CMake throws an error.

**Comment by kostrzewa** [comment]
Ah, the quotes were the problem, cheers Dean!

**Comment by mathiaswagner** [comment]
Looks good. I assume that this works for you so I am happy to merge it and whenever we need a better solution can go on from here.

File: CMakeLists.txt [hunk 1]
```diff
@@ -230,12 +232,18 @@ endif()
...
-  find_library(MAGMA ${MAGMA_LIBRARIES} PATH ${MAGMA_LIBRARY_DIRS})
+  if("${QUDA_MAGMAHOME}" STREQUAL "")
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(MAGMA  magma)
+    include_directories(SYSTEM ${MAGMA_INCLUDEDIR})
+    message("${MAGMA_INCLUDEDIR}")
+    find_library(MAGMA ${MAGMA_LIBRARIES} PATH ${MAGMA_LIBRARY_DIRS})
+  else()
+    # prefer static library
+    FIND_LIBRARY(MAGMA_LIB libmagma.a magma ${QUDA_MAGMAHOME}/lib/)
```
Comment (kostrzewa) [inline CMakeLists.txt]: @maddyscientist @mathiaswagner 
Sorry, I messed up here, this should be ```MAGMA``` and not ```MAGMA_LIB```. Will submit hotfix.

**Pull request merged**